code/ui/generalhud/playerinfo/PlayerInfo.cs

```diff
@@ -65,8 +69,15 @@ public override void Tick()
                 }
                 else
                 {
-                    _roleLabel.Text = $"{player.Role.Name.ToUpper()}";
-
+                    switch (_currentRole) {
+                        case InnocentRole:
+                            _roleLabel.Text = lang.GetTranslation("ROLE_NAME_INNOCENT");
+                            break;
+
+                        case TraitorRole:
+                            _roleLabel.Text = lang.GetTranslation("ROLE_NAME_TRAITOR");
+                            break;
+                    }
```

Comment:
That definitely should be handled with `$"ROLE_NAME_{_currentRole.Name.ToUpper()}"` or similar OR in the role instead of here :)

Reply:
It might be worth just have methods inside of the ILanguage class that handles this logic.

I.e

```csharp
string GetRoleTranslation(Role role) {
	...
}
```

Thoughts?

Reply:
@mzegar
With English - yes. But, for example, in Russian you can't just say "Он был Предатель", instead you say "Он был Предател**ем**"

Reply:
$"ROLE_NAME_{_currentRole.Name.ToUpper()}" is more appropriate. But I think switch is faster and more readable.